vllm/model_executor/layers/sampler.py

```diff
@@ -473,34 +485,71 @@ def _multinomial(
     probs: torch.Tensor,
     num_samples: int,
     seq_groups: Optional[List[SequenceGroupToSample]] = None,
+    is_fallback: bool = False,
 ) -> torch.Tensor:
-    if num_samples > 1:
-        # This is equivalent to torch.repeat_interleaved (which also
-        # forces a GPU<->CPU sync).
-        # This allows us to do sampling with replacement by creating
-        # num_samples copies of each row in the tensor, and then
-        # batch sampling the resulting tensor.
-        probs = probs[:, None, :].expand(probs.shape[0], num_samples,
-                                         probs.shape[1]).contiguous().view(
-                                             -1, probs.shape[1])
-    q = torch.empty_like(probs)
+    if num_samples > 1 and not is_fallback:
+        probs = probs.repeat_interleave(num_samples, dim=0)
```

Comment:
Can we use the shape of probs to determine whether to apply `repeat_interleave` instead of introducing `is_fallback`?

Reply:
This is now resolved by using flashinfer's new kernels as fallback.